ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -75,6 +75,27 @@ class InputTrace {
   }
 };
 
+// Function virtual addresses observed in the input trace.
+// Maps a function symbol name to a virtual address.
+class FuncAddrs {
+  map<string, uintptr_t> Map;
+
+public:
+  FuncAddrs(char **FuncNames, size_t *VAddrs, size_t Len) {
+    for (size_t I = 0; I < Len; I++) {
+      Map.insert(pair<char *, size_t>(FuncNames[I], VAddrs[I]));
+    }
+  }
+
+  // Lookup the address of the specified function name or return 0 on failure.
```

Comment:
If this is returning an address should it return `void */NULL` ?

Reply:
Good question.

Although they are broadly equivalent, it may be better as a `void *`, as we eventually pass this do an LLVM API function which accepts `void *`.